draft-ietf-tls-tls13.md

```diff
@@ -4375,6 +4375,89 @@ currently deployed and properly configured TLS implementations.
 %%### Ticket Establishment
 %%### Updating Keys
 
+# State Machine
+
+This section provides a summary of the legal state transitions for the
+client and server handshakes.  State names (in all capitals, e.g.,
+START) have no formal meaning but are provided for ease of
+comprehension.  Messages which are sent only sometimes are indicated
+in [].
+
+## Client
+
+~~~~
+          +-----> START
+     Recv |        |
+    Hello |        | Send ClientHello
+    Retry |        v
+  Request +---- WAIT_SH            
+                   | Recv ServerHello
+                   V                 
+                WAIT_EE            
+                   | Recv EncryptedExtensions         
+          +--------+--------+          
+Using PSK |                 | Using Certificate             
+          |                 v
+          |            WAIT_CERT_CR
+          |        Recv |       | Recv CertificateRequest
+          | Certificate |       v
+          |             |    WAIT_CERT
+          |             |       | Recv Certificate
+          |             v       v
+          |              WAIT_CV
+          |                 | Recv CertificateVerify
+          +> WAIT_FINISHED <+
+                   | Recv Finished
+                   |
+                   | [Send EndOfEarlyData]
+                   | [Send Certificate [+ CertificateVerify]]
```

Comment:
The way that the client rejects client auth is by sending an empty certificate message.